backend/stakepoold/server_test.go

```diff
@@ -30,28 +33,39 @@ var (
 func init() {
 
 	c = &appContext{
+		userTicketsNew: make(map[string]UserTickets),
 		userTicketsOld: make(map[string]UserTickets),
 		sendTickets:    sendTicketsDummy,
 	}
 
 	// Create a pool of tickets around expected size
 	ticketCount := 49000
-	userCount := uint16(10000)
+	userCount := 10000
 	for i := 0; i < ticketCount; i++ {
-		b := randomBytes(2)
-		uid := binary.LittleEndian.Uint16(b) % userCount
+		b := randomBytes(4)
+		uid := int(binary.LittleEndian.Uint32(b)) % userCount
+		msa := strconv.Itoa(uid | 1<<31)
+		ticket := &chainhash.Hash{b[0], b[1], b[2], b[3]}
 
-		// ugly on memory but works
-		ut := c.userTicketsOld[strconv.Itoa(int(uid))]
+		// We abuse memody pretty badly here but it is test.
+		// OLD: use MultiSigAddress as the key
+		ut := c.userTicketsOld[msa]
 		ut.Userid = int64(uid)
-		ut.MultiSigAddress = strconv.Itoa(int(uid))
-		ut.Tickets = append(ut.Tickets, &chainhash.Hash{b[0], b[1]})
-		c.userTicketsOld[strconv.Itoa(int(uid))] = ut
+		ut.MultiSigAddress = msa
+		ut.Tickets = append(ut.Tickets, ticket)
+		c.userTicketsOld[msa] = ut
+
+		// NEW: use ticket as the key
+		tS := ticket.String()
+		utNew := c.userTicketsNew[tS]
+		utNew.Userid = int64(uid)
+		utNew.MultiSigAddress = msa
+		utNew.Tickets = append(utNew.Tickets, ticket)
+		c.userTicketsNew[tS] = utNew
```

Comment:
Switching `c.userTicketsNew` to a map with tickets as the key and then embedding arrays of tickets doesn't seem right, or maybe I'm just confused.

I was thinking to leave the multisig-keyed map, but change the definition of `UserTickets` to:

```golang
type UserTickets struct {
	Userid          int64
	MultiSigAddress string
	Tickets         map[*chainhash.Hash]struct{}
}
```

That way you can just replace

```if !sliceContains(ctx.userTickets[msa].Tickets, ticket) {```

with

```if _, ok := ctx.userTickets[msa].Tickets[ticket]; !ok {```